projects/igniteui-angular/src/lib/tree/tree.component.ts

```diff
@@ -501,9 +502,13 @@ export class IgxTreeComponent implements IgxTree, OnInit, AfterViewInit, OnDestr
     private subToChanges() {
         this.unsubChildren$.next();
         const toBeSelected = [...this.forceSelect];
-        requestAnimationFrame(() => {
+        if(this.platform.isBrowser) {
+            requestAnimationFrame(() => {
+                this.selectionService.selectNodesWithNoEvent(toBeSelected);
+            });
+        } else {
```

Comment:
Is it needed to set the selection on the server? I assume that when the view hydrates it would go through the same logic and apply selection if needed, so maybe the `else` handling here is redundant.